src/storage/scripting.h

```diff
@@ -101,4 +103,66 @@ void SHA1Hex(char *digest, const char *script, size_t len);
 int RedisMathRandom(lua_State *l);
 int RedisMathRandomSeed(lua_State *l);
 
+/// ScriptFlags turn on/off constraints or indicate properties in Eval scripts and functions
+///
+/// Note: The default for Eval scripts are different than the default for functions(default is 0).
+/// As soon as Redis sees the #! comment, it'll treat the script as if it declares flags, even if no flags are defined,
+/// it still has a different set of defaults compared to a script without a #! line.
+/// Another difference is that scripts without #! can run commands that access keys belonging to different cluster hash
+/// slots, but ones with #! inherit the default flags, so they cannot.
+enum ScriptFlags : uint64_t {
```

Comment:
enum class?

Reply:
I understand that `enum class` has better type safety, but when used here, it is more like a bitwise `uint64_t`. If using `enum class` requires the use of `static_cast`, it feels very troublesome. Do you have any better suggestions?